partiql-tests-data/success/syntax/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,402 @@
+// SQL style
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - No Second Precision ",
+  statement: "TIMESTAMP '2023-06-01 00:00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Arbitrary Second Precision",
+  statement: "TIMESTAMP '2023-06-01 00:00:00.00000000000",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision = Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision < Digit in Second Fraction",
+  statement: "TIMESTAMP(0) '2023-06-01 00:00:00.0000",
+  assert: [
+    {
+      result: SyntaxSuccess
+    },
+  ]
+}
+
+{
+  name: "TIMESTAMP WITHOUT TIME ZONE - SQL Style - Precision > Digit in Second Fraction",
```

Comment:
The special characters used in the names (i.e. `=`, `<`, `>`) could have a naming collision depending on how the conformance test runner deals with special characters. Might be better to use alphanumeric characters here.